### PR TITLE
Fix some points for improvement

### PR DESCRIPTION
**Pull request by reprimande**
I've fixed some points for example improving the lack of import library.
And I wonder some behaviors as below.

### `takeTo`

Currently some tests are failed

1. test case 'n is bigger than sum'

Test case
```
    await t.test('n is bigger than sum', () => {
        const actual = takeTo(15, [1, 2, 3]);
        assert.deepStrictEqual(actual, [1, 2, 3, 1, 2, 3, 1, 2]);
    });
```

Result
```
  ✖ n is bigger than sum (3.782167ms)
    AssertionError [ERR_ASSERTION]: Expected values to be strictly deep-equal:
    + actual - expected

      [
        1,
        2,
        3,
    -   1,
    -   2,
    -   3,
    -   1,
    -   2
      ]
        at TestContext.<anonymous> (file:///Users/n/src/github.com/reprimande/array-toolkit/test/arrayTransformations.mjs:375:16)
        at Test.runInAsyncScope (node:async_hooks:206:9)
        at Test.run (node:internal/test_runner/test:580:25)
        at Test.start (node:internal/test_runner/test:493:17)
        at TestContext.test (node:internal/test_runner/test:129:20)
        at TestContext.<anonymous> (file:///Users/n/src/github.com/reprimande/array-toolkit/test/arrayTransformations.mjs:373:13)
        at async Test.run (node:internal/test_runner/test:581:9)
        at async Test.processPendingSubtests (node:internal/test_runner/test:325:7) {
      generatedMessage: true,
      code: 'ERR_ASSERTION',
      actual: [ 1, 2, 3 ],
      expected: [
        1, 2, 3, 1,
        2, 3, 1, 2
      ],
      operator: 'deepStrictEqual'
    }
```
Is my expectation is not correct?

2. `n is 0`

Test Case
```
    await t.test('n is 0', () => {
        const actual = takeTo(0, [1, 2, 3]);
        assert.deepStrictEqual(actual, []);
    });
```

Result
```
  ✖ n is 0 (0.176542ms)
    AssertionError [ERR_ASSERTION]: Expected values to be strictly deep-equal:
    + actual - expected

    + [
    +   0
    + ]
    - []
        at TestContext.<anonymous> (file:///Users/n/src/github.com/reprimande/array-toolkit/test/arrayTransformations.mjs:380:16)
        at Test.runInAsyncScope (node:async_hooks:206:9)
        at Test.run (node:internal/test_runner/test:580:25)
        at Test.start (node:internal/test_runner/test:493:17)
        at TestContext.test (node:internal/test_runner/test:129:20)
        at TestContext.<anonymous> (file:///Users/n/src/github.com/reprimande/array-toolkit/test/arrayTransformations.mjs:378:13)
        at async Test.run (node:internal/test_runner/test:581:9)
        at async Test.processPendingSubtests (node:internal/test_runner/test:325:7) {
      generatedMessage: true,
      code: 'ERR_ASSERTION',
      actual: [ 0 ],
      expected: [],
      operator: 'deepStrictEqual'
    }
```
I guess the result is empty is correct. Isn't it?

3. `n is -1`

Test Case
```
    await t.test('n is -1', () => {
        const actual = takeTo(-1, [1, 2, 3]);
        assert.deepStrictEqual(actual, []);
    });
```

Result
```
  ✖ n is -1 (0.545542ms)
    AssertionError [ERR_ASSERTION]: Expected values to be strictly deep-equal:
    + actual - expected

    + [
    +   '-1': NaN
    + ]
    - []
        at TestContext.<anonymous> (file:///Users/n/src/github.com/reprimande/array-toolkit/test/arrayTransformations.mjs:386:16)
        at Test.runInAsyncScope (node:async_hooks:206:9)
        at Test.run (node:internal/test_runner/test:580:25)
        at Test.start (node:internal/test_runner/test:493:17)
        at TestContext.test (node:internal/test_runner/test:129:20)
        at TestContext.<anonymous> (file:///Users/n/src/github.com/reprimande/array-toolkit/test/arrayTransformations.mjs:384:13)
        at async Test.run (node:internal/test_runner/test:581:9)
        at async Test.processPendingSubtests (node:internal/test_runner/test:325:7) {
      generatedMessage: true,
      code: 'ERR_ASSERTION',
      actual: [ '-1': NaN ],
      expected: [],
      operator: 'deepStrictEqual'
    }
```
I guess it's also correct that the result is empty.

### `loopTo`

It's almost same as `takeTo`. 

```
  ✖ targetLength is 0 (0.34875ms)
    AssertionError [ERR_ASSERTION]: Expected values to be strictly deep-equal:
    + actual - expected

    + [
    +   0
    + ]
    - []
        at TestContext.<anonymous> (file:///Users/n/src/github.com/reprimande/array-toolkit/test/arrayTransformations.mjs:411:16)
        at Test.runInAsyncScope (node:async_hooks:206:9)
        at Test.run (node:internal/test_runner/test:580:25)
        at Test.start (node:internal/test_runner/test:493:17)
        at TestContext.test (node:internal/test_runner/test:129:20)
        at TestContext.<anonymous> (file:///Users/n/src/github.com/reprimande/array-toolkit/test/arrayTransformations.mjs:409:13)
        at async Test.run (node:internal/test_runner/test:581:9)
        at async Test.processPendingSubtests (node:internal/test_runner/test:325:7) {
      generatedMessage: true,
      code: 'ERR_ASSERTION',
      actual: [ 0 ],
      expected: [],
      operator: 'deepStrictEqual'
    }

  ✖ targetLength is -1 (0.179875ms)
    AssertionError [ERR_ASSERTION]: Expected values to be strictly deep-equal:
    + actual - expected

    + [
    +   '-1': NaN
    + ]
    - []
        at TestContext.<anonymous> (file:///Users/n/src/github.com/reprimande/array-toolkit/test/arrayTransformations.mjs:417:16)
        at Test.runInAsyncScope (node:async_hooks:206:9)
        at Test.run (node:internal/test_runner/test:580:25)
        at Test.start (node:internal/test_runner/test:493:17)
        at TestContext.test (node:internal/test_runner/test:129:20)
        at TestContext.<anonymous> (file:///Users/n/src/github.com/reprimande/array-toolkit/test/arrayTransformations.mjs:415:13)
        at async Test.run (node:internal/test_runner/test:581:9)
        at async Test.processPendingSubtests (node:internal/test_runner/test:325:7) {
      generatedMessage: true,
      code: 'ERR_ASSERTION',
      actual: [ '-1': NaN ],
      expected: [],
      operator: 'deepStrictEqual'
    }

  ✖ array is empty (0.072833ms)
    RangeError [Error]: Invalid array length
        at Function.from (<anonymous>)
        at buildArray (file:///Users/n/src/github.com/reprimande/array-toolkit/dataToArray.mjs:16:18)
        at loopTo (file:///Users/n/src/github.com/reprimande/array-toolkit/arrayTransformations.mjs:211:25)
        at TestContext.<anonymous> (file:///Users/n/src/github.com/reprimande/array-toolkit/test/arrayTransformations.mjs:422:24)
        at Test.runInAsyncScope (node:async_hooks:206:9)
        at Test.run (node:internal/test_runner/test:580:25)
        at Test.start (node:internal/test_runner/test:493:17)
        at TestContext.test (node:internal/test_runner/test:129:20)
        at TestContext.<anonymous> (file:///Users/n/src/github.com/reprimande/array-toolkit/test/arrayTransformations.mjs:421:13)
        at async Test.run (node:internal/test_runner/test:581:9)
```

Please try to run test 😃

### CODE REVIEW

File: arrayTransformations.mjs [hunk 1]
```diff
@@ -7,15 +7,16 @@
 // --------------------------------------------------------------------------
 
 //requiring it locally:
-// let buildArray = require('./dataToArray.js').buildArray;
+import { buildArray } from './dataToArray.mjs';
+import * as R from 'ramda';
```
Comment (reprimande) [inline arrayTransformations.mjs:12]: These library is necessary in this code. This code couldn't be excuted.

File: arrayTransformations.mjs [hunk 2]
```diff
@@ -7,15 +7,16 @@
...
-  * @param {number} number - the desired length of adjusted array
-  * @param {array} array - the array to modify
-  * @example console.log(resizeArray(5, [0,1,2])) // [0, 1, 2, 0, 1]
-*/
+ * Set a size of the array and the array will be expanded or shrunk to fix that size. To expand it will just loop the array:
+ * Does the same thing as adjustArrayLength but does it differently
+ * @param {number} number - the desired length of adjusted array
+ * @param {array} array - the array to modify
+ * @example console.log(resizeArray(5, [0,1,2])) // [0, 1, 2, 0, 1]
+ */
```
Comment (reprimande) [inline arrayTransformations.mjs:20]: Sorry, I use prettier (https://prettier.io/) for auto formatting on save. So this kind of change is caused by it.

File: arrayTransformations.mjs [hunk 3]
```diff
@@ -108,104 +109,104 @@ export function scaleToRange(inputArray, inputMin, inputMax, outputMin, outputMa
...
-  * @example console.log(pickN(3, [1,2,3,4,5,6,7,8,9])) //[9, 5, 8]
-*/
+ * Picks multiple items randomly in an array.
+ * @param {number} n - the desired number of item that are picked
+ * @param {array} inputArray - the array to pick items from
+ * @example console.log(pickN(3, [1,2,3,4,5,6,7,8,9])) //[9, 5, 8]
+ */
 export function pickN(n, inputArray) {
-    return this.buildArray(n, (i) => pick(inputArray));
+    return buildArray(n, (i) => pick(inputArray));
```
Comment (reprimande) [inline arrayTransformations.mjs:142]: In JavaScript, `this` is the parent of the instance. In this library's case, every function is top level.  These functions don't have parent instance.

File: test/arrayTransformations.mjs [hunk 4]
```diff
@@ -21,62 +21,27 @@ import {
...
+    removeAtIndex,
+} from '../arrayTransformations.mjs';
 
 // Note: It looks that the result of `resizeArray` is same as `adjustArrayLength`.
 test('resizeArray', async (t) => {
     await t.test('same size', () => {
-        const result = resizeArray(3, [1, 2, 3]);
-        assert.equal(result.length, 3);
-        assert.deepStrictEqual(result, [1, 2, 3]);
+        const actual = resizeArray(3, [1, 2, 3]);
```
Comment (reprimande) [inline test/arrayTransformations.mjs:30]: It just renames variable. Sorry for so many change...

**Comment by renickbell** [comment]
thanks! i'll look at this today.

**Pull request closed without merging**